magellan-library/src/main/java/com/wealthfront/magellan/navigation/NavigableCompat.kt

```diff
@@ -3,4 +3,6 @@ package com.wealthfront.magellan.navigation
 import com.wealthfront.magellan.core.Displayable
 import com.wealthfront.magellan.lifecycle.LifecycleAware
 
-public interface NavigableCompat : LifecycleAware, Displayable
+public interface NavigableCompat : LifecycleAware, Displayable {
+  public val currentNavigable: NavigableCompat get() = this
```

Comment:
hmm why is this method on Navigable instead of Navigator? Please don't tell me it's because of LazySetNavigator 😩

Reply:
oh I see - it lets us avoid reflection when sifting thru backstacks. We can just treat all Navigables equally i.e. without checking for Navigators only

Reply:
oh, and this also allows the application developer to define current screen when it's otherwise undefined. Example: a Step containing two "sibling" Journeys doesn't have a well-defined "current" screen

Reply:
Exactly. I could imagine a master-detail arrangement, where an application developer would have a better sense of what the "current" navigable is